Source/WebKit/UIProcess/WebNavigationState.cpp

```diff
@@ -102,16 +102,28 @@ RefPtr<API::Navigation> WebNavigationState::takeNavigation(uint64_t navigationID
     return m_navigations.take(navigationID);
 }
 
-void WebNavigationState::didDestroyNavigation(uint64_t navigationID)
+void WebNavigationState::didDestroyNavigation(WebCore::ProcessIdentifier processID, uint64_t navigationID)
 {
     ASSERT(navigationID);
-
-    m_navigations.remove(navigationID);
+    auto it = m_navigations.find(navigationID);
+    if (it != m_navigations.end() && (*it).value->processID() == processID)
```

Comment:
Does this need to check both, or can it just take the navigation with the id and assert that the process identifier is correct?

Reply:
Oh it definitely needs to check both. When we process-swap, the process we swap away from will send us an IPC to destroy the navigation and we want to ignore it since the ownership of the navigation has been transferred to a new process.